### PR TITLE
Refine Docker Build Process for Multiplatform Support (incl. M3, AMD, ARM/32)

### PR DESCRIPTION
**Pull request by Freezor**
### Description:

This pull request refines the Docker build process to enhance multiplatform support, including architectures M3, AMD, and ARM/32. It also updates the multi-architecture build configuration to align with project requirements.

## Motivation and Context

The aim of this change is to improve the Docker build pipeline by expanding multiplatform support to include M3 architecture alongside existing AMD and ARM/32 architectures. This enhancement ensures broader compatibility and optimized deployment capabilities across diverse hardware environments.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

The changes have been locally tested with Docker builds targeting M3, AMD, and ARM/32 architectures. Automated tests are integrated into the CI/CD pipeline to ensure build integrity across platforms.

## Screenshots (if appropriate):

N/A

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: .github/workflows/build-and-publish-docker-images.yml [hunk 1]
```diff
@@ -24,56 +46,84 @@ jobs:
...
-          load: true
           file: ./src/docker/Dockerfile-AasxServerBlazor
-          tags: adminshellio/aasx-server-blazor-for-demo:latest
+          platforms: |
+            linux/amd64
+            linux/arm/v6
+            linux/arm/v7
+            linux/arm64
+          tags: |
+            adminshellio/aasx-server-blazor-for-demo:v3latest
```
Comment (AlexanderWollbrink) [inline .github/workflows/build-and-publish-docker-images.yml:60]: we can take out v3latest here, I think

File: .github/workflows/build-and-publish-docker-images.yml [hunk 2]
```diff
@@ -24,56 +46,84 @@ jobs:
...
-          file: ./src/docker/Dockerfile-AasxServerBlazor-arm32
-          tags: adminshellio/aasx-server-blazor-for-demo-arm32:latest
+          file: ./src/docker/Dockerfile-AasxServerAspNetCore
+          platforms: |
+            linux/amd64
+            linux/arm/v6
+            linux/arm/v7
+            linux/arm64
+          tags: |
+            adminshellio/aasx-server-aspnetcore-for-demo:v3latest
```
Comment (AlexanderWollbrink) [inline .github/workflows/build-and-publish-docker-images.yml:76]: we can take out v3latest here too then

Comment (Freezor) [reply .github/workflows/build-and-publish-docker-images.yml]: changed to 'latest'

File: .github/workflows/build-and-publish-docker-images.yml (hunk 1 above)
Comment (Freezor) [reply .github/workflows/build-and-publish-docker-images.yml]: changed to 'latest'

**Pull request merged**